Mtm 59369/change logs automation

shaikhahme opened the pull request:
1. Added new workflow to trigger updating changelogs daily
2.  Updated changelogs.yml to better better log number of changed files
3. Removed redundant trigger workflows
4. Added different version formats handling
5. Replaced Environment variables and update repo link for c8y-iot-build-pipeline repository from forked to original
6. Replace manual token generation with actions/create-github-app-token@v1

shaikhahme committed:
MTM-59369: adding Workflow and complimentary python script

shaikhahme committed:
MTM-59369:

shaikhahme committed:
MTM-59369: removing dates to test files

shaikhahme committed:
Revert "MTM-59369: removing dates to test files"

This reverts commit 70344f9bd85bbf0e80ec1561c1db4fde7a9bdd38.

shaikhahme committed:
MTM-59369: removing dates to test files

shaikhahme committed:
Revert "MTM-59369: removing dates to test files"

This reverts commit 5bfc326470c796cfa84d62ea1b2832b2957ceba2.

shaikhahme committed:
Merge pull request #10 from shaikhahme/change-logs-automation

Change logs automation

shaikhahme committed:
MTM-59369: cleaning up code and removing redundant files

eliasweingaertner commented:
@shaikhahme I really like this, as it looks simple and straightforward. Two questions:

a) How did you test it?
b) What happens if the python script runs into an error? Does the pipeline execution stop as expected?

shaikhahme committed:
MTM-59369: added explicit error handling in python script

shaikhahme committed:
MTM-59369: Testing error handling when buildartifact not present

shaikhahme committed:
MTM-59369: Testing error handling when buildartifact not present

shaikhahme committed:
MTM-59369: Testing error handling when helm_charts_data.json has a bad key

shaikhahme committed:
MTM-59369: Testing error handling when helm_charts_data.json has a bad key

shaikhahme committed:
MTM-59369: Testing error handling when helm_charts_data.json has a bad key

shaikhahme committed:
MTM-59369: Testing error handling when helm_charts_data.json has a json structure

shaikhahme committed:
MTM-59369: correcting Error type in python script and preparing for testing automation

shaikhahme committed:
MTM-59369: testing new forked repo post migration and cumulocity action for token generation

shaikhahme committed:
MTM-59369: testing new forked repo post migration and cumulocity action for token generation

shaikhahme committed:
MTM-59369: updating environment variables and secrets

shaikhahme committed:
MTM-59369: updating environment variables and secrets

shaikhahme committed:
MTM-59369: updated type script code

shaikhahme committed:
MTM-59369: updated type script code

shaikhahme committed:
MTM-59369: updated python code to enforce unique output

shaikhahme committed:
Merge pull request #13 from shaikhahme/MTM-59369/testing-change-logs-automation

Tested and Added:
1. Successfully creates PR for updated components:
https://github.com/shaikhahme/c8y-docs/actions/runs/11212598996
 Created PR: https://github.com/shaikhahme/c8y-docs/pull/12/files

2. Error Handling: workflows fail when an error occurs with an appropriate error message 
eg:
Intentional FAIL when buildartifact or helm_charts_data.json files not found: 
https://github.com/shaikhahme/c8y-docs/actions/runs/11144924947/job/30973302337#step:6:14

Intentional fail when wrong key used: 
https://github.com/shaikhahme/c8y-docs/actions/runs/11145129833/job/30973992903#step:6:18

Intentional fail when improper json structure present:
https://github.com/shaikhahme/c8y-docs/actions/runs/11145195684/job/30974213508#step:6:14

shaikhahme commented:
> @shaikhahme I really like this, as it looks simple and straightforward. Two questions:
> 
> a) How did you test it? b) What happens if the python script runs into an error? Does the pipeline execution stop as expected?

a) I tested the scripts locally and also in GitHub actions
b) yes the pipeline execution stops, I also added error messages to make to help.

you can see the workflow logs for different test scenarios from the comment in b821be05e1c89a8b28aff927cd3b279920ea8d9a

eliasweingaertner commented:
@shaikhahme asked me to mark this ready for review, as he has problems with Github.

reey commented on .github/workflows/daily-changelog-dates-trigger.yml:
```diff
@@ -0,0 +1,82 @@
+      - name: Checkout c8y-iot-build-pipeline repository
+        uses: actions/checkout@v4
+        with:
+          repository: ${{ vars.FORKED_REPO }}
```
Why a `Forked` repo?

shaikhahme replied:
@reey That is because I can't install an app on the https://github.com/Cumulocity-IoT/c8y-iot-build-pipeline repo. And I needed the app for authentication. 

or did I misunderstand how to use apps for authentication?

reey replied:
Okay, but in the future it won't be forked, right? So the name of the variable should not include `FORKED_` prefix..

shaikhahme replied:
resolved in 1dc3707a3b05d181699b7313746dc6e46318e04e

reey commented on .github/workflows/update-changelog-dates.yml:
It seems a few of these changes are taken from: https://github.com/SoftwareAG/c8y-docs/pull/2382
Any reason why you've skipped some of these changes?

shaikhahme replied:
@reey No particular reason, I wanted to keep the changes to update-changelog-dates.yml minimal to keep testing easier, but I'll add the other changes too. thanks for catching it!

shaikhahme replied:
resolved in a9b317e67f8967cfc8ee386f785248dd75dd51ce

reey commented on scripts/get-component-details-from-chart.py:
Any reason for rewriting https://github.com/SoftwareAG/c8y-docs/pull/2382/files#diff-4f3d2a35fb5ebe6c2282a25deabc2a777ccf68218838e1f164c47f8c97ae257fR64-R85 as a python script?
I at least think that we have more developers familiar with shell scripts compared to python to be honest..

shaikhahme replied:
@reey this was a decision taken after talking to @eliasweingaertner , who suggested using python because we have more developers familiar with python than bash.

eliasweingaertner replied:
Whoops. That was a misunderstanding: When @shaikhahme told me he was working on a bash script, I said that I'd prefer python. I didn't intend to have a working implementation replaced.

@reey Still, I believe the Python script is more readable. Are you okay with keeping that as well?

reey replied:
I'm fine with a python script as long as someone maintains it (because I won't)..

Oezge-Akin replied:
@shaikhahme are you doing sth. additional than what is done in https://github.com/Cumulocity-IoT/c8y-docs/pull/2382/files#diff-4f3d2a35fb5ebe6c2282a25deabc2a777ccf68218838e1f164c47f8c97ae257fR64-R85

If it's same or easy to extend, then let's use the existing one to not duplicate things. 
If it is more complex then we can continue with python since it has good util libraries and it can make development/maintenance easier.

shaikhahme replied:
Understood, since the python script has almost the same functionality as the bash script, I will use bash script.

shaikhahme replied:
resolved in 1dc3707a3b05d181699b7313746dc6e46318e04e

reey commented on add-release-dates/index.ts:
```diff
@@ -32,19 +34,24 @@ async function processFile(filePath: string): Promise<boolean> {
-      "is not a valid semver. Skipping.."
-    );
+
+  if (!valid(data.version) && data.version.split('.').length==4){
+    const parts = data.version.split('.');
+    const normalizedVersion = `${parts[0]}${parts[1]}.${parts[2]}.${parts[3]}`;
+    data.version=normalizedVersion ;
+    }
```
Can you log a warning here that we are trying to convert the none semver version into semver?

shaikhahme replied:
resolved in 1dc3707a3b05d181699b7313746dc6e46318e04e

shaikhahme committed:
MTM-59369: updating changelogs.yml

shaikhahme committed:
MTM-59369: Resolving comments
1. Replacing python with bash script
2. renaming FORKED_REPO variable
3. Added warning log when correcting version format

shaikhahme committed:
MTM-59369: fixing error in bash script

shaikhahme committed:
MTM-59369:
1. Using configured environment and its variables
2. Using github action for token generation

shaikhahme committed:
MTM-59369:
1. Using configured environment and its variables
2. Using github action for token generation

shaikhahme committed:
MTM-59369:
fixing typo

shaikhahme committed:
MTM-59369: Using github action for token generation

shaikhahme committed:
MTM-59369: fixing typo

shaikhahme committed:
Merge branch 'Cumulocity-IoT:develop' into MTM-59369/change-logs-automation

shaikhahme committed:
Merge branch 'Cumulocity-IoT:develop' into develop

shaikhahme committed:
Merge branch 'develop' into MTM-59369/change-logs-automation

shaikhahme committed:
Merge branch 'develop' into MTM-59369/change-logs-automation

reey commented on .github/workflows/daily-changelog-dates-trigger.yml:
```diff
@@ -0,0 +1,76 @@
+        with:
+          app-id: ${{ vars.APP_ID }}
+          private-key: ${{ secrets.APP_PRIVATE_KEY }}
+          owner: cumulocity-iot
+          repositories: c8y-iot-build-pipeline
```
Should allow us to avoid any changes in case of the next GitHub migration.
The `repositories` parameter is not required as the GitHub app we are using anyhow only has access to this single repo.
```suggestion
          owner: ${{ github.repository_owner }}
```

reey commented on add-release-dates/index.ts:
```diff
@@ -32,19 +34,25 @@ async function processFile(filePath: string): Promise<boolean> {
-      "is not a valid semver. Skipping.."
-    );
+
+  if (!valid(data.version) && data.version.split('.').length==4){
```
#trivial
```suggestion
  if (!valid(data.version) && data.version.split('.').length==4) {
```

reey commented on add-release-dates/index.ts:
```diff
@@ -32,19 +34,25 @@ async function processFile(filePath: string): Promise<boolean> {
+    data.version=normalizedVersion ;
+    }
+
+  if(!valid(data.version)){
```
#trivial
```suggestion
  }

  if(!valid(data.version)) {
```

reey commented on add-release-dates/index.ts:
```diff
@@ -32,19 +34,25 @@ async function processFile(filePath: string): Promise<boolean> {
+    console.debug("Version in file: ",filePath,"is not a valid semver. Skipping..");
     return false;
-  }
+    }
```
#trivial
```suggestion
  }
```

reey commented on scripts/get-component-details-from-chart.py:
It seems you've decided to use the shell script instead now? If so, this file is obsolete and can be removed, right?

shaikhahme replied:
thanks for catching that. Resolved in ed86ba313b520018693f4edcfdefa896db1709ea

shaikhahme committed:
Update .github/workflows/daily-changelog-dates-trigger.yml

Co-authored-by: Tristan Bastian <tristan-c.bastian@gmx.de>

shaikhahme committed:
Update add-release-dates/index.ts

Co-authored-by: Tristan Bastian <tristan-c.bastian@gmx.de>

shaikhahme committed:
Update add-release-dates/index.ts

Co-authored-by: Tristan Bastian <tristan-c.bastian@gmx.de>

shaikhahme committed:
Update add-release-dates/index.ts

Co-authored-by: Tristan Bastian <tristan-c.bastian@gmx.de>

shaikhahme committed:
Merge remote-tracking branch 'origin/MTM-59369/change-logs-automation' into MTM-59369/change-logs-automation

shaikhahme committed:
MTM-59369: removing redundant python script

shaikhahme committed:
Merge branch 'develop' into MTM-59369/change-logs-automation
